### ISSUE TITLE
Custom rule warning appears in line 1

### ISSUE THREAD

**Issue by eli-gini** [realm/SwiftLint#4719]
Using SwiftLint 0.50.3, installed using CocoaPods.
I added a custom rule to find boolean variables/constants that aren't starting with the word 'is'. Any files in the project containing an error are flagged at the beginning of the file instead of the actual line where the error occurs.

<img width="962" alt="Screenshot 2023-01-24 at 11 11 19" src="https://user-images.githubusercontent.com/94921809/214252965-fed9e738-0423-4de6-aa24-5ae05344eea1.png">

* Paste your configuration file:

```yml
disabled_rules:
 - colon
 - trailing_whitespace
 - mark
 - line_length
 - identifier_name
 - comma
 - type_name
 - indentation_width

opt_in_rules:
- indentation_width
 
excluded:
 - Pods/
 - OneNotificationService/
 - ONETests/
 - OneArticlesWidget/
 - Frameworks/

custom_rules:
        single_line_vertical_spacing:
            include: "*.swift"
            regex: '\}[^\n]*\n[^\n]*(class|struct|enum|extension|func)'
            name: "Single line vertical space"
            message: "Expected single whitespace"
            severity: warning
                
        boolean_declaration_prefix:
            include: "*.swift"
            regex: '^(?!override).*(var|let)\s((?!is)(\S)+)(((\s*Bool)(\?|\!)?)|\s(=\s(true|false))){1,2}(\s\{)?$'
            name: "Boolean property declaration"
            message: "Boolean type must start with 'is'"
```

* Are you using [nested configurations]?
  No

* Which Xcode version are you using?
 Xcode 14.0.1
Build version 14A400

Other rules are working as expected. Regex was tested on regex editors online and seems to work fine.
Any ideas?

**Comment by SimplyDanny** [comment]
`.` in custom regular expressions is configured to match also newlines by default. That's why your rule matches right from the beginning of the file. You can disable this behavior by adding `(?-s)` to your regular expression.

Let me know if this works for you.

**Comment by SimplyDanny** [comment]
This issue has been marked as stale because it has not seen any activity in the last 4 months. It is going to be closed soon if it stays inactive.
